openmdao/surrogate_models/kriging.py

```diff
@@ -150,6 +156,14 @@ def _calculate_reduced_likelihood_params(self, thetas=None):
         params['U'] = U
         params['Vh'] = Vh
 
+        # Newly added for EGOLF
+        one = np.ones([self.n_samples,1])
+        R_inv = Vh.T.dot(np.dot(np.diag(inv_factors),U.T)) #Use einsum in the future release for efficiency
```

Comment:
Good idea. But for now, let's keep the krigging file clean and I will close this request. I am talking to the OpenMDAO team tomorrow. They will do the refactoring and code optimization of the algorithm I am trying to integrate.